mii/grpc_related/restful_gateway.py

```diff
@@ -0,0 +1,62 @@
+import time
+import threading
+import mii
+from flask import Flask, request
+from flask_restful import Resource, Api
+from werkzeug.serving import make_server
+from mii.constants import RESTFUL_GATEWAY_SHUTDOWN_TIMEOUT, RESTFUL_API_PATH
+from google.protobuf.json_format import MessageToJson
+
+
+def shutdown(thread):
+    time.sleep(RESTFUL_GATEWAY_SHUTDOWN_TIMEOUT)
+    thread.server.shutdown()
+
+
+def createRestfulGatewayApp(task, mii_config, server_thread):
+    # client must be thread-safe
+    client = mii.MIIClient(task, "localhost", mii_config.port_number)
+
+    class RestfulGatewayService(Resource):
+        def __init__(self):
+            super().__init__()
+
+        def post(self):
+            data = request.get_json()
+            kwargs = data["kwargs"] if "kwargs" in data else {}
+            result = client.query(data["request"], **kwargs)
+            return MessageToJson(result)
+
+    app = Flask("RestfulGateway")
+
+    @app.route("/terminate", methods=['GET'])
+    def terminate():
+        # Need to shutdown *after* completing the request
+        threading.Thread(target=shutdown, args=(server_thread, )).start()
+        return "Shutting down RESTful API gateway server"
+
+    api = Api(app)
+    path = '/{}/{}'.format(RESTFUL_API_PATH, task)
```

Comment:
I think it would be better to use the `deployment_name` rather than the `task` so we can have 2 MII servers running with the same task.

Reply:
I agree. I will change as proposed.